src/averageAges.js

```diff
@@ -55,10 +92,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothers = people.filter(person => person.sex === FEMALE);
+
+  const whoHasMothers = people.filter((person) => {
+    return onlyWithSon
+      ? (person.sex === MALE && person.mother !== null)
+      : person.mother !== null;
```

Comment:
```suggestion
      ? (person.sex === MALE && person.mother)
      : person.mother;
```
You don't need to compare it with null, just check if exists.
Also, it's better to name peopleWithMothers or something like this instead of whoHasMothers.